servlet/src/main/scala/org/http4s/servlet/syntax/ServletContextSyntax.scala

```diff
@@ -46,15 +48,25 @@ final class ServletContextOps private[syntax] (val self: ServletContext) extends
   ): ServletRegistration.Dynamic =
     mountHttpApp(name, service.orNotFound, mapping, dispatcher)
 
+  def mountRoutes[F[_]: Async](
+      name: String,
+      service: HttpRoutes[F],
+      mapping: String = "/*",
+      dispatcher: Dispatcher[F],
+      asyncTimeout: Duration = defaults.ResponseTimeout,
+  ): ServletRegistration.Dynamic =
+    mountHttpApp(name, service.orNotFound, mapping, dispatcher, asyncTimeout)
+
   def mountHttpApp[F[_]: Async](
       name: String,
       service: HttpApp[F],
       mapping: String = "/*",
       dispatcher: Dispatcher[F],
+      asyncTimeout: Duration = defaults.ResponseTimeout,
```

Comment:
This is _source_ compatible, but not _binary_ compatible.  Default arguments are applied at compile time, but at link time, they don't exist.  So this is good, but we need to keep the old version too.  I think the old copy needs to:

1. be deprecated, so something screams if we try to call it 
2. be `private[servlet]`, so we don't get in the way of callers of this version.
3. remove the default argument to `mapping`, because two overloads can't have default arguments
4. call the new one like `mountHttpApp(name, service, mapping, defaults.ResponseTimeout)`

Reply:
Oh I understand now. I'll close this pr and open a new one to target `series/0.23`.